src/python/modules/ensembl/compara/citest/tabletest.py

```diff
@@ -0,0 +1,398 @@
+        obj_request = text(str_request)
+        obj_connection = obj_connection_engine.connect()
+        obj_result = obj_connection.execute(obj_request)
+        for row in obj_result:
+            str_table_name = row[tabl_name]
+            lst_pipeline_table_names.append(str_table_name)
+        obj_connection.close()
+        return lst_pipeline_table_names
+
+    def run_tests(self):
+        """Main function running the table tests
+
+        For each tables intialised in initialise_tests this method will
+        compare the number of row between the reference and target db
+        and if some table has group by test and/or column test it will carry on
+        these tests.
+            Args:
+                None
+            Returns:
+                None
+        """
+        for dic_tabl_test in self.table_tests:
+            str_table_name = dic_tabl_test["name"]
+            int_line_variation = dic_tabl_test["variation"]
+            lst_group_by = dic_tabl_test["lst_group_by"]
+            lst_columns = dic_tabl_test["columns"]
+
+            #First it compare the number of row between the ref and target db
+            self._test_table_number_row(str_table_name, int_line_variation)
+
+            #test the group if defined in json
+            if not lst_group_by == []:
+                self._test_table_number_row_by_group(str_table_name, lst_group_by)
+
+            #test field if present in json
+            if not lst_columns == []:
+                self._test_table_by_column(str_table_name, lst_columns)
+
+    def _test_table_number_row(self, str_table_name, flt_variation):
+        """Function comparing the number of row between a table in the ref and target DB
+
+        For each tables intialised in initialise_tests this method will
+        compare the number of row between the reference and target db
+        and if some table has group by test and/or column test it will carry on
+        these tests. The result of the test is store as dictionary
+        in self.test_results attribute
+            Args:
+                str_table_name: the name of the table to test (str)
+                flt_variation: the variation (flt)
+            Returns:
+                None
+        """
+        result = {}
+        #do the check for the given table
+        int_nb_row_ref = self._get_number_row(self.ref_connection_engine, str_table_name)
+        int_nb_row_tar = self._get_number_row(self.target_connection_engine, str_table_name)
+        int_variation = int(int_nb_row_ref*flt_variation)
+        result["succes"] = 0
+        if abs(int_nb_row_ref - int_nb_row_tar) <= int_variation: # if the test is succesful
+            result["succes"] = 1
+
+        #populate results with test information
+        str_info = "ref:" + str(int_nb_row_ref)+"|"+"tar:"+str(int_nb_row_tar)
+        str_info = str_info + "|var:"+str(int_nb_row_ref - int_nb_row_tar)
+        str_info = str_info + "|exp:"+str(int_variation)
```

Comment:
This should be single statement. Maybe you can do something like
```suggestion
        titles = ["ref", "tar", "var", "exp"]
        values = [int_nb_row_ref, int_nb_row_tar, int_nb_row_ref - int_nb_row_tar, int_variation]
        str_info = "|".join("{}:{}".format(*t) for t in zip(titles, values))
```

Reply:
good idea, I will do it